Python/iSense.py

```diff
@@ -0,0 +1,310 @@
+        self.__url = dictionary['url']
+        self.__path = dictionary['path']
+        self.__created_at = dictionary['createdAt']
+        self.__field_count = dictionary['fieldCount']
+        self.__data_point_count = dictionary['datapointCount']
+        self.__display_url = dictionary['displayURL']
+        self.__data = dictionary['data']
+
+
+    # Refresh pulls down the data sets latest data from isense
+    def refresh(self):
+        self.__init__(self.data)
+        return
+
+    def getId(self):
+        return self.__id
+
+    def getName(self):
+        return self.__name
+
+    def getPath(self):
+        return self.__path
+
+    def getCreationTime(self):
+        return self.__created_at
+
+    def getFieldCount(self):
+        return self.__field_count
+
+    def getDataPointCound(self):
+        return self.__data_point_count
+
+    def getDisplayUrl(self):
+        return self.__display_url
+
+    def getDataByField(self, field):
+        # return all data related to a field
+        data = []
+
+        for point in self.__data:
+            data.append(point[str(field.getId())])
+
+        return data
+
+    def getData(self):
+        return self.__data
+
+    # given another data object it will append it to this data set
+    def appendData(self, data_set_id, data_dictionary, credentials):
+
+        # Append the new data to the data set on iSENSE
+        if credentials.isUser():
+            response = iSenseAPI().appendDataSet(credentials.getUsername(), credentials.getPassword(),
+                data_set_id, data_dictionary)
+        elif credentials.isKey():
+            response = iSenseAPI().appendDataSet(credentials.contrib_key(),
+                data_set_id, data_dictionary)
+        else:
+            response = INVALID_CREDENTIALS 
+
+        # After the data is uploaded refresh this data set object so it has the latest data 
+        refresh()
+
+        # Response is a dictionary. It can be used to check if the post request was successful.
+        # It also will contain any errors the server may have returned.
+        return response
+
+   
+
+
+class Field(object):
+    def __init__(self, field_id, name, field_type, unit="", restrictions=[]):
+        self.__id = field_id
+        self.__name = name
+        self.__type = field_type
+        self.__restrictions = restrictions
+        self.__unit = unit
+
+    def getType(self):
+        return self.__type
+
+    def getName(self):
+        return self.__name
+
+    def getId(self):
+        return self.__id
+
+    def getRestrictions(self):
+        return self.__restrictions
+
+    def getUnit(self):
+        return self.__unit
+
+# Credentials requires either (username and password) OR (key and contrib_name)
+class Credentials(object):
+    def __init__(self, username=None, password=None, 
+                 contrib_key=None, contrib_name=None):
+
+        self.__user = False
+        self.__key = False
+
+        if username and password:
```

Comment:
I think it should default to username and password. A user can do everything you can do with a key and more. If a key allowed you to do different things that a user could not then I would throw an error but because a user can do everything a key can do it makes sense to just ignore the key if they provide both.